### ISSUE TITLE
Settings not being persistet across restarts

### ISSUE THREAD

**Issue by AuspeXeu** [pi-hole/docker-pi-hole#668]
This is a: Run Issue

## Details
When setting custom upstream DNS servers, these settings reset every time I restart the container or do an update with docker-compose.

## How to reproduce the issue 

1. Environment data
  * Operating System: CentOS7
  * Hardware: PC
  * Kernel Architecture: x86/amd64
  * Docker Install Info and version: 
    - Software source: official docker-ce
  * Hardware architecture: x86

2. docker-compose.yml contents
```
version: '2.2'
services:
  pihole:
    container_name: pihole
    image: pihole/pihole:latest
    ports:
      - "53:53/tcp"
      - "53:53/udp"
      - "67:67/udp"
      - "80:80/tcp"
      - "443:443/tcp"
    environment:
      TZ: 'Europe/Berlin'
      WEBPASSWORD: 'xxxxxxxxxxxxxxxx'
    # Volumes store your data between container upgrades
    volumes:
      - './etc-pihole/:/etc/pihole/'
      - './etc-dnsmasq.d/:/etc/dnsmasq.d/'
    dns:
      - 192.168.0.1
      - 1.1.1.1
      - 8.8.8.8
    restart: unless-stopped
```

## These common fixes didn't work for my issue

- [x] I have tried removing/destroying my container, and re-creating a new container
- [x] I have tried fresh volume data by backing up and moving/removing the old volume data
- [x] I have tried running the stock `docker run` example(s) in the readme (removing any customizations I added)
- [ ] I have tried a newer or older version of Docker Pi-hole (depending what version the issue started in for me)
- [x] I have tried running without my volume data mounts to eliminate volumes as the cause

**Comment by JuniperMS** [comment]
I came here to report the same thing.  You beat me to it!

**Comment by millecentdix** [comment]
Hi,
Same issue for me at the last restart on Debian 10/OMV 5.

**Comment by jaymz668** [comment]
is this the same issue?
https://github.com/pi-hole/docker-pi-hole/issues/604

**Comment by millecentdix** [comment]
Yes, it is the same issue.

**Comment by aheid** [comment]
I got the same issue, Armbian buster with Linux 5.7.15-sunxi, Docker version 19.03.12. Had it with 5.0 image and now 5.1.2 image. When I recreate the container my settings are lost. 

When I hit save in the web UI, I can see the changes reflected in the `etc-pihole/setupVars.conf` file on the host, so the volume seems to be mapped correctly.

**Comment by lhzw** [comment]
It's so disgusting, the rubbish s6-init script will overwrite the dsnmasq config file on each container's restart, totally a shit!!!
It overwrites the config file in the container. It overwrites the volumed config file!!!
It's unacceptable, can you accept your system reset everything just after you reboot your system?? What's in your head. It's so a wasting of our invaluable time.
Did you test your image before you release it? Yes, it works with the default configs, but we can never modify the configuration file.
I want to modify something else more than the dns upstream.

`chown: cannot access '/etc/pihole/dhcp.leases': No such file or directory
  [i] Existing dnsmasq.conf found... it is not a Pi-hole file, leaving alone!
  [i] Copying 01-pihole.conf to /etc/dnsmasq.d/01-pihole.conf...^MESC[K  [✓] Copying 01-pihole.conf to /etc/dnsmasq.d/01-pihole.conf
Setting password: VndHCJ6m
`

**Comment by damajor** [comment]
It seems the same issue applies for any settings in setupVars.conf

**Comment by nathang21** [comment]
You can set both of those values via Docker environment variables, just add `DNS1` and `DNS2` to your compose file (see Readme for details)

**Comment by jaymz668** [comment]
you may be able to, but you shouldn't have to. Settings should carry forward

**Comment by damajor** [comment]
should ? That is vague...

I can confirm that at least QUERY_LOGGING is not taken in account and it is overwritten with the default value each time you restart.

**Comment by MrDomoArigato** [comment]
I'm having the same issue on MacOS. The hope is to eventually move to a pi but my company wants to see a working product before I move over. Is this an issue on raspbian? I also only see one client and am wondering if this is a related issue.

**Comment by TheGITofTeo997** [comment]
Hello, i have opened #719 because i've been facing the exact same issue.

**Comment by PromoFaux** [comment]
No need for a new issue - lets keep things in one place.

**Comment by PromoFaux** [comment]
Closed in place of the pinned #720
